connaisseur/flask_server.py

```diff
@@ -18,17 +24,32 @@
 """
 
 
+@APP.errorhandler(AlertSendingError)
+def handle_alert_sending_failure(err):
+    logging.error(err.message)
+    return (str(err.message), 500)
+
+
+@APP.errorhandler(ConfigurationError)
+def handle_alert_config_error(err):
+    logging.error(err.message)
+    return (str(err.message), 500)
+
+
 @APP.route("/mutate", methods=["POST"])
 def mutate():
     """
     Handles the '/mutate' path and accepts CREATE and UPDATE requests.
     Sends its response back, which either denies or allows the request.
     """
+
     admission_request = request.json
     try:
         validate(admission_request)
         response = admit(admission_request)
     except BaseConnaisseurException as err:
+        if call_alerting_on_request(admission_request, admitted=False):
```

Comment:
i think by now we should be nesting the exceptions. something like this:

```python
try:
        validate(admission_request)
        response = admit(admission_request)
    except Exception as err:
        if call_alerting_on_request(admission_request, admitted=False):
            send_alerts(admission_request, admitted=False)
        
        if isinstance(err, BaseConnaisseurException):
            err_log = str(err)
            msg = err.user_msg
        elif isinstance(err, UnknownVersionError):
            msg, err_log = str(err), str(err)
        else:
            err_log = str(traceback.format_exc())
            msg = "unknown error. please check the logs."
        
        logging.error(err_log)
        return jsonify(
            get_admission_review(
                admission_request.get("request", {}).get("uid"),
                False,
                msg=msg,
                detection_mode=DETECTION_MODE,
            )
        )
    if call_alerting_on_request(admission_request, admitted=True):
        send_alerts(admission_request, admitted=True)
    return jsonify(response)
```

Alternatively we could also go for the Flask errorhandlers which might be even sexier. :thinking:

Reply:
you can add the comment `# pylint: disable=no-member` in the line where pylint complains, so that the error is only disabled there and not everywhere

Reply:
Loos like we're running into this pylint stuff a lot. Is there a way to configure strictness leverls? Maybe we need a lower one.

Reply:
> Unfortunately, pylint complains about the line `msg = err.user_msg` if `err` is of instance `BaseConnaisseurException`, becasue it got to know `err` as plain `Exception` which does not have the `user_msg` property.

stupid linter.

Reply:
I just skimmed its docs and it doesn't seem that there's a strictness level option